x/concentrated-liquidity/math/tick.go

```diff
@@ -174,6 +185,6 @@ func CalculatePriceToTick(price sdk.Dec) (tickIndex sdk.Int) {
 	ticksToBeFulfilledByExponentAtCurrentTick := osmomath.BigDecFromSDKDec(price.Sub(currentPrice)).Quo(currentAdditiveIncrementInTicks)
 
 	// Finally, add the ticks we have passed from the completed geometricExponent values, as well as the ticks we have passed in the current geometricExponent value
-	tickIndex = ticksPassed.Add(ticksToBeFulfilledByExponentAtCurrentTick.SDKDec().TruncateInt())
+	tickIndex = ticksPassed.Add(ticksToBeFulfilledByExponentAtCurrentTick.SDKDec().RoundInt())
```

Comment:
asking to learn: why are we making this change? What's our decision for the truncate int vs round int?

Reply:
This is the core of the change. Due to residue in intermediary calculations, currently, we get off-by-one errors.

Imagine a price of 50000
```
>>> Decimal("50000").sqrt()
Decimal('223.6067977499789696409173669')
>>> math.pow(Decimal("50000").sqrt(), 2)
50000.0
```

With our sdk.Dec:
```
sqrt = "223.606797749978969640917367"
pow(sqrt, 2) = "50000.000000000000000000000057"
```

We then use that value in some of the intermediary calculations, and the residue gets the following result:
```
call priceBigDec.Sub(currentPrice).Quo(currentAdditiveIncrementInTicks).String()
floor("-4999999.999999999999999999994300000000000000") = -4999999
```

When it should have been:
```
50000.000000000000000000000000 - 100000.000000000000000000000000 = -50000.000000000000000000000000
floor("-50000.000000000000000000000000") = -50000.000000000000000000000000
```

Reply:
I think it should be fine. This behavior occurs due to squaring a fixed point real number. The magnitude of the change should be bounded by the existing test vectors around max spot price where it rounds correctly right now.

So I believe there should be no case where this breaks.